### PR TITLE
feat: new action for automerging dependabot and pre-commit.ci PRs

### PR DESCRIPTION
**Pull request by RobPasMue**
WIP... needs testing.

### CODE REVIEW

File: doc/source/housekeeping-actions/examples/hk-automerge-prs-basic.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+hk-automerge-prs:
+  name: "Automerging dependabot and pre-commit.ci PRs"
+  runs-on: ubuntu-latest
+  # Only runs if we are on a PR
+  if: github.event_name == 'pull_request'
+  steps:
+    - name: "Automerging dependabot and pre-commit.ci PRs"
+      uses: ansys/actions/hk-automerge-prs@{{ version }}
+      with:
+        approver: 'mygithubid'
```
Comment (jorgepiloto) [inline doc/source/housekeeping-actions/examples/hk-automerge-prs-basic.yml:10]: We could suggest here the bot.

Comment (RobPasMue) [reply doc/source/housekeeping-actions/examples/hk-automerge-prs-basic.yml:10]: good point yeah - let me add the expected defaults. I just wanted to let it open to configuration

**Comment by RobPasMue** [comment]
Action is working perfectly - see

https://github.com/ansys/pyansys-tools-report/pull/240 for implementation
https://github.com/ansys/pyansys-tools-report/pull/242 for dependabot PR
https://github.com/ansys/pyansys-tools-report/pull/244 for pre-commit.ci PR

File: hk-automerge-prs/action.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Verify allowed PRs (dependabot and pre-commit.ci)
+      shell: bash
+      run: |
+        if [[ ${{ github.head_ref }} == 'dependabot/'* ]] || [[ ${{ github.head_ref }} == 'pre-commit-ci-update-config' ]]; then
```
Comment (jorgepiloto) [inline hk-automerge-prs/action.yml:43]: Any change we can actually check for the author? Just worried about someone opening a branch starting with "dependabot/" or "pre-commit" to auto-merge malicious stuff in a repo where externals are authorized for CI/CD runs.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Just left some concern about checking for PR name rather than checking the author.

File: hk-automerge-prs/action.yml (hunk 2 above)
Comment (RobPasMue) [reply hk-automerge-prs/action.yml:43]: Good catch... I can try and check for the author as well.

Comment (RobPasMue) [reply hk-automerge-prs/action.yml:53]: See https://github.com/ansys/actions/pull/652/commits/e3e3529d362ed45297d35fd672a20e78e9f41396

**Review by SMoraisAnsys — COMMENTED** [review]
Can we have a flavor (input) to not allow automatically minor bumps ?
That way one could keep reading new PRs with new features while automatically accepting patch releases.

**Comment by RobPasMue** [comment]
> Can we have a flavor (input) to not allow automatically minor bumps ? That way one could keep reading new PRs with new features while automatically accepting patch releases.

Sure..

**Comment by RobPasMue** [comment]
> Can we have a flavor (input) to not allow automatically minor bumps ? That way one could keep reading new PRs with new features while automatically accepting patch releases.

See https://github.com/ansys/actions/pull/652/commits/a1659cb8b62a4ebba62afe3ac44944a1982a62e7

**Review by SMoraisAnsys — APPROVED** [review]
LGTM, I just discovered the `dependabot/fetch-metadat` project and it seems great !
There is a thing that we might consider to leverage (for example to expose something relating to fixing a security issue), see `alert-lookup``in https://github.com/dependabot/fetch-metadata?tab=readme-ov-file#usage-instructions

**Review by jorgepiloto — APPROVED** [review]
Thanks, @RobPasMue. This is going to be really useful.

**Pull request merged**